design/014/THREAT_MODEL.md

```diff
@@ -0,0 +1,329 @@
+- ✅ CA private key never distributed to provider namespaces
+- ✅ Strict RBAC limits access to ESO namespace
+- ⚠️ Consider using external system for CA key storage (KMS/HSM)
+
+**Impact:** Critical - full compromise of provider authentication
+
+**Residual Risk:** Medium (depends on cluster hardening)
+
+---
+
+### T3: Rogue Provider Registration
+
+**Threat:** Attacker deploys malicious provider and gets ESO to communicate with it.
+
+**Attack Vector:**
+- Deploy service with `external-secrets.io/provider: "true"` label
+- ESO generates certificates for the rogue service
+- Create Provider resource pointing to rogue service
+- Intercept or manipulate secret requests
+
+**Mitigation:**
+- ✅ Provider-specific configuration still required (can't serve AWS secrets without AWS credentials)
+- ✅ RBAC controls who can create Provider resources
+- ⚠️ Consider admission webhook to validate Provider resources
+- ⚠️ Monitor for unexpected service labels
+
+**Impact:** Low - attacker can't access external secrets without proper credentials. Provider impersonation may be a DOS vector.
+
+**Residual Risk:** Low
+
+---
+
+### T4: Provider Impersonation
+
+**Threat:** Attacker impersonates legitimate provider to return malicious secrets.
+
+**Attack Vector:**
+- Steal provider TLS certificates
+- Deploy service at same address as legitimate provider
+- Respond to ESO requests with malicious data
+
+**Mitigation:**
+- ✅ mTLS with certificate validation prevents impersonation
+- ✅ Certificates tied to specific DNS names
+- ✅ NetworkPolicies can restrict communication paths
+- ⚠️ Certificate rotation limits window of compromised certs
+
+**Residual Risk:** Low
+
+---
+
+### T5: Supply Chain Attack via Provider Dependencies
+
+**Threat:** Malicious code in provider dependencies exfiltrates secrets or credentials.
+
+**Attack Vector:**
+- Compromised provider SDK
+- Vulnerable dependencies in provider binary
+- Backdoored base images
+
+**Mitigation:**
+- ✅ Providers built with minimal dependencies
+- ✅ Each provider isolated - compromise doesn't affect others
+- ⚠️ Dependency scanning and SBOM generation recommended
+- ⚠️ Image signing and verification
+- ⚠️ Regular security audits of provider code
+
+**Impact:** High - provider can access external secret systems and have a lot of RBAC permissions within Kubernetes.
+
+**Residual Risk:** Medium (ongoing supply chain risk)
+
+---
+
+### T6: Denial of Service via Certificate Exhaustion
+
+**Threat:** Attacker creates many labeled services to exhaust ESO Core resources.
+
+**Attack Vector:**
+- Deploy numerous services with provider label
+- ESO generates certificates for all services
+- Resource exhaustion prevents legitimate operations
+
+**Mitigation:**
+- ⚠️ Rate limiting on certificate generation (not yet implemented)
+- ⚠️ RBAC limits who can create services with eso-provider label
+- ⚠️ Monitoring and alerting on certificate generation rate
+- ⚠️ Resource quotas on ESO Core pod
```

Comment:
another mitigation would be delegating this to a new pod (e.g. cert-controller). Exhaustion would prevent certificate renewal and new providers onboarding, but not day-to-day operations.